test/acceptance/features/steps/steps.py

```diff
@@ -145,27 +151,39 @@ def db_instance_is_running(context, db_name):
 @when(sbr_is_applied_step)
 def sbr_is_applied(context):
     sbr_yaml = context.text
-    sbr = ServiceBindingRequest()
-    if context.__contains__("nodejs_app"):
-        application = context.nodejs_app
-        context.nodejs_app_original_generation = application.get_observed_generation()
-        context.nodejs_app_original_pod_name = application.get_running_pod_name()
+    namespace = context.namespace
+    sbr = ServiceBindingRequest(namespace.name)
+    if context.__contains__("application"):
+        application = context.application
+        context.application_original_generation = application.get_observed_generation()
+        if context.application_type.__contains__("knative"):
+            context.application_original_rev_name = application.get_rev_name()
+        else:
+            context.application_original_pod_name = application.get_running_pod_name()
     sbr.create(sbr_yaml) | should.be_truthy.desc("Service Binding Request Created")
 
-
 # STEP
+
+
 @then(u'application should be re-deployed')
 def then_application_redeployed(context):
-    application = context.nodejs_app
-    application.get_redeployed_pod_name(context.nodejs_app_original_pod_name) | should_not.be_none.desc(
-        "There is a running pod of the application different from the original one before redeployment.")
+    application = context.application
+    if context.application_type.__contains__("knative"):
```

Comment:
Do you think we can make use of switch statement here? so in future, if any new application comes instead of extending the if-else ladder we can add a new case.

Reply:
there is no switch in python, have to implement a dict function to handle that.

Reply:
See if it makes sense make the changes or will keep as it is for now.

Reply:
will keep as is now

Reply:
Okay, you can resolve this conversation.